keycloak/keycloak-operator
# INTLY-8271 - skip backups if no instances to backup are found

## JIRA ID

https://issues.redhat.com/browse/KEYCLOAK-14409

## Additional Information

When no matching realms or keycloaks were found for resources (clients, users, backups) then the reconciliation loop was skipped and an error was logged. This also prevented the finalizers on a resources from getting updated and thus, orphaned resources could no longer be deleted.

This change removes the error when no matches are found and instead treats that case as an error for backups exclusively (because backups actually require an instance to be present).

## Verification Steps

1. Deploy keycokak, a realm, a client, a user and a backup (you can use the examples)
2. Delete the realm CR
3. Delete the keycloak CR
4. Delete the user CR: this should still work and you should see a message in the operator logs saying the finalizer got removed.
5. Delete the client CR: this should still work and you should see a message in the operator logs saying the finalizer got removed
6. Delete the backup CR: this should work. No finalizers involved.

## Checklist:
